tests/unit/test_routing_manager.cpp

```diff
@@ -4287,6 +4287,57 @@ void TestBorderRoutingProcessPlatfromGeneratedNd(void)
         VerifyOmrPrefixInNetData(localOmr, /* aDefaultRoute */ false);
     }
 
+    // 6. Replace a prefix, on some platforms, there might be no messages to deprecate the old prefix, instead, they
+    // send new prefixes directly.
+    //    In this case, we still use the old prefix as long as the old prefix preferred lifetime is not exceeded, and
+    //    replace it with the new prefix when expired.
+    Log("6. Replace prefix.");
+    {
+        Ip6::Prefix raPrefix    = PrefixFromString("2001:db8:1:2::", 64);
+        Ip6::Prefix newRaPrefix = PrefixFromString("2001:db8:3:4::", 64);
+
+        SendRouterAdvertToBorderRoutingProcessIcmp6Ra({Pio(raPrefix, kValidLitime, kPreferredLifetime)});
+
+        sExpectedRios.Add(raPrefix);
+        sExpectedRios.Clear();
+        AdvanceTime(10000);
+
+        VerifyPdOmrPrefix(raPrefix);
+        VerifyOmrPrefixInNetData(raPrefix, /* aDefaultRoute */ false);
+
+        AdvanceTime(1000000);
+        VerifyPdOmrPrefix(raPrefix);
+
+        // Send new prefix without deprecating old prefix.
+        // The old prefix should be preferred for another (1800 - 10 - 1000) = 790s
+        SendRouterAdvertToBorderRoutingProcessIcmp6Ra({Pio(newRaPrefix, kValidLitime, kPreferredLifetime)});
+        AdvanceTime(500000);
+        VerifyPdOmrPrefix(raPrefix);
+
+        AdvanceTime(300000);
```

Comment:
Suggest using `AdvanceTime(300 * 1000)` instead of `(300000)` here and in other places in the newly added test case (6). It makes it similar to case (5) and would be easier to read. We can also consider changing all of these it a future PR?